### PR TITLE
RSS for Tag show with image correction

### PR DESCRIPTION
**Pull request by SidharthBansal**
Make sure these boxes are checked before your pull request is ready to be reviewed and merged. Thanks!

* [x] all tests pass -- `rake test:all`
* [x] code is in uniquely-named feature branch, and has been rebased on top of latest master (especially if you've been asked to make additional changes)
* [x] pull request is descriptively named with #number reference back to original issue

Please be sure you've reviewed our contribution guidelines at https://publiclab.org/wiki/contributing-to-public-lab-software

We have a loose schedule of reviewing and pulling in changes every Tuesday and Friday, and publishing changes on Fridays. Please alert developers on plots-dev@googlegroups.com when your request is ready or if you need assistance.

Thanks!

### CODE REVIEW

**Comment by SidharthBansal** [comment]
@jywarren I have made the rss for show action in accordance with the show_for_author rss, as suggested by Sagar Preet. So that  images will work properly.
Also, I checked the previous rss on production, there were some small bugs created by me which are resolved in this commit. (-->It was rendering the markdown text as simple text)
Please review it

**Comment by SidharthBansal** [comment]
@jywarren have you added something in settings for plots2? As soon as I open a new pr. I get a message that you have labeled the pr as in progress.

**Comment by ViditChitkara** [comment]
Happened with me too! That's pretty cool though!!

**Comment by SidharthBansal** [comment]
Yeah, pretty cool stuff. @jywarren can easily filter out labeled ready ones first and review them. And we can easily see what is pending on our side.

**Comment by jywarren** [comment]
Hmm, whoa, no i did not change anything... not sure how that's happening! But i like your idea of filtering with labels. I put up a post at #2231 to think about ways we can make this workflow smoother. Good idea!

**Comment by jywarren** [comment]
For this one, could I get a review by @sagarpreet-chadha ? Thank you both!!

**Comment by jywarren** [comment]
@sagarpreet-chadha are you interested in being a reviewer, part of our reviewers group? I had somehow thought you already were but will invite you now in case you're interested. Thank you! 

https://publiclab.org/wiki/developers#Reviewers

**Comment by SidharthBansal** [comment]
Yeah I also was willing to get a review first from him and then from you as he is doing the rss stuff. But I guess he is not in the reviewers team so he cant review my pr. Can you please add him so that he can approve or suggest changes to this pr as well.
Thanks

File: app/views/tag/rss.rss.builder [hunk 1]
```diff
@@ -22,10 +25,12 @@ xml.rss :version => "2.0" do
...
           xml.pubDate     node.created_at.to_s(:rfc822)
         end
-        #xml.link        url_for :only_path => false, :controller => 'notes', :action => 'show', :id => node.nid
         xml.link        "https://" + request.host.to_s + node.path
-        #xml.image "//#{ request.host }/"+node.main_image.path(:default) if node.main_image
-        xml.description auto_link(node.latest.render_body, :sanitize => false)
+        if node.main_image
+          xml.description  {  xml.cdata!("<img src='#{node.main_image.path(:default)}' alt='#{node.main_image.title}'><p>#{auto_link(node.latest.render_body, :sanitize => false)}</p>")  }
+        else
+          xml.description  {  xml.cdata!("<img src='https://i.publiclab.org/system/images/photos/000/000/354/medium/Boots-ground-02.png' alt='PublicLab'><p>#{auto_link(node.latest.render_body, :sanitize => false)}</p>")  }
```
Comment (sagarpreet-chadha) [inline app/views/tag/rss.rss.builder]: Hi , can you add this image [https://publiclab.org/system/images/photos/000/023/444/original/Screenshot_20180204-101546_2.png](https://publiclab.org/system/images/photos/000/023/444/original/Screenshot_20180204-101546_2.png) as used in #2221 to maintain uniformity . Awesome !

File: app/views/tag/rss_for_tagged_with_author.rss.builder [hunk 2]
```diff
@@ -27,9 +27,9 @@ xml.rss :version => '2.0', 'xmlns:atom' => 'http://www.w3.org/2005/Atom' do
         end
         xml.link        "https://" + request.host.to_s + node.path
         if node.main_image
-          xml.description  {  xml.cdata!("<img src='#{node.main_image.path(:default)}' alt='#{node.main_image.title}'>")  }
+          xml.description  {  xml.cdata!("<img src='#{node.main_image.path(:default)}' alt='#{node.main_image.title}'><p>#{auto_link(node.latest.render_body, :sanitize => false)}</p>")  }
         else
-          xml.description  {  xml.cdata!("<img src='https://i.publiclab.org/system/images/photos/000/000/354/medium/Boots-ground-02.png' alt='PublicLab'><p>#{body}</p>")  }
+          xml.description  {  xml.cdata!("<img src='https://i.publiclab.org/system/images/photos/000/000/354/medium/Boots-ground-02.png' alt='PublicLab'><p>#{auto_link(node.latest.render_body, :sanitize => false)}</p>")  }
```
Comment (sagarpreet-chadha) [inline app/views/tag/rss_for_tagged_with_author.rss.builder]: Here also :-)

**Comment by sagarpreet-chadha** [comment]
Thank you @jywarren for adding me to reviewers group 😄 !

Hi @SidharthBansal ,  the changes seems perfect to me 💯 . Thank you !

**Comment by SidharthBansal** [comment]
@sagarpreet-chadha changes done

File: app/views/tag/rss.rss.builder [hunk 3]
```diff
@@ -1,19 +1,22 @@
...
+
       body = node.body
-      body = "<img src='"+node.main_image.path(:default)+"'/><br /> "+node.body if node.main_image
-    
-      xml.item do
-        xml.title       node.title
-        xml.author      node.author.name
-        if node.power_tag('date') != ''
+      uname = node.author.username
+       email = node.author.email
```
Comment (jywarren) [inline app/views/tag/rss.rss.builder]: Oh, sorry - note that I changed this so we don't publish peoples' email addresses without asking them -- would you mind incorporating these changes too? https://github.com/publiclab/plots2/pull/2221/files

Thanks and apologies, i should have noted this earlier!

**Comment by SidharthBansal** [comment]
@jywarren changes done
Thanks

**Comment by jywarren** [comment]
Awesome, merging now. Great workflow and collaboration on this one! 🎉 👥

**Comment by SidharthBansal** [comment]
Thanks

**Pull request merged**